### PR TITLE
Create markdown link check task

### PR DESCRIPTION
**Pull request by RDIL**
This is a WIP that checks to see if all links work or not.  It may be a breaking change

### CODE REVIEW

File: .vsts-ci/linkcheck.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: PR-$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:.rr)
+trigger:
+  branches:
+    include:
+    - master
+    - release*
+  paths:
+    exclude:
+    - /src/*
```
Comment (iSazonov) [inline .vsts-ci/linkcheck.yml]: Why not use "include *.md" instead of "exclude" ?

Comment (RDIL) [reply .vsts-ci/linkcheck.yml]: Good point, I will.

Comment (TravisEz13) [reply .vsts-ci/linkcheck.yml]: Please don't.. wildcards only work if they are the last character

File: .vsts-ci/linux.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ pr:
   paths:
     include:
     - '*'
+    - /.vsts-ci/linkcheck.yml
```
Comment (TravisEz13) [inline .vsts-ci/linux.yml]: shouldn't this be an exclude?

File: .vsts-ci/linkcheck.yml (hunk 1 above)
Comment (TravisEz13) [reply .vsts-ci/linkcheck.yml]: Can you add this to the spelling?   I can do a rename later.

File: .vsts-ci/linux.yml (hunk 2 above)
Comment (RDIL) [reply .vsts-ci/linux.yml]: I’ll work on this later

**Comment by RDIL** [comment]
K

File: .vsts-ci/linkcheck.yml (hunk 1 above)
Comment (RDIL) [reply .vsts-ci/linkcheck.yml]: Doing now.

**Comment by RDIL** [comment]
@TravisEz13 added to `spelling` CI task.

**Comment by RDIL** [comment]
The check is working!!!!!! :D

**Comment by RDIL** [comment]
Sadly the check doesn't return a code of 1 if dead links are found but it still works!

File: .vsts-ci/spelling.yml [hunk 3]
```diff
@@ -35,6 +35,15 @@ phases:
       sudo npm install -g markdown-spellcheck@0.11.0
     displayName: Install mdspell
     condition: succeededOrFailed()
+    
+  - bash: |
+      sudo npm install -g markdown-link-check@3.7.2
```
Comment (RDIL) [inline .vsts-ci/spelling.yml]: Warning: this version must stay static on `3.7.2`!

File: .vsts-ci/spelling.yml [hunk 4]
```diff
@@ -44,3 +53,8 @@ phases:
       mdspell '**/*.md' '!**/Pester/**/*.md' --ignore-numbers --ignore-acronyms --report --en-us;
     displayName: Test Spelling
     condition: succeededOrFailed()
+    
+  - bash: |
+      find . -name \*.md -exec markdown-link-check {} \;
```
Comment (RDIL) [inline .vsts-ci/spelling.yml]: This `find` method is the most efficient I could find.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
Two issues I left comments for.  Add a comment about pinning versions and fix code factor issues.

File: .vsts-ci/spelling.yml (hunk 3 above)
Comment (RDIL) [reply .vsts-ci/spelling.yml]: Is it just a hashtag?

**Comment by RDIL** [comment]
Okay.

File: .vsts-ci/spelling.yml (hunk 3 above)
Comment (TravisEz13) [reply .vsts-ci/spelling.yml]: yes

**Comment by RDIL** [comment]
It is now fixed!!

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Thank @RDIL

**Comment by RDIL** [comment]
Thank you!

**Pull request merged**